fontir/src/depgraph.rs

```diff
@@ -227,23 +306,41 @@ impl DepGraph {
         new_files.iter().for_each(|p| paths.push(p));
 
         // Anything change 'round here?
+        let mut path_states = HashMap::new();
+        let mut changed = HashSet::new();
         for path in paths {
             let path_state = PathState::of(path)?;
             let has_changed = self
                 .entries
                 .get(path)
                 .map(|prior_state| *prior_state != path_state)
                 .unwrap_or(true);
+            path_states.insert(path, path_state);
 
             if has_changed {
-                changes.push(Change {
-                    path: path.to_owned(),
-                    path_state,
-                });
+                // Add path and, recursively, things that depend on it
+                let mut to_add = Vec::new();
+                to_add.push(path);
+                while let Some(path) = to_add.pop() {
+                    if !changed.insert(path) {
+                        continue;
+                    }
+                    if let Some(deps) = self.dependencies.get(path) {
+                        deps.iter().for_each(|e| to_add.push(e));
+                    }
```

Comment:
do we not need to check for cycles?

Reply:
oh sorry that's exactly what what `if !changed.insert(path) continue;` does... 😅 
HashSet.insert returns false when it already contains the value